internal/reader/ccip.go

```diff
@@ -168,26 +164,24 @@ func (r *CCIPChainReader) MsgsBetweenSeqNums(
 func (r *CCIPChainReader) NextSeqNum(
 	ctx context.Context, chains []cciptypes.ChainSelector,
 ) ([]cciptypes.SeqNum, error) {
-	if err := r.validateReaderExistence(r.destChain); err != nil {
-		return nil, err
+	cfgs, err := r.getSourceChainsConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get source chains config: %w", err)
 	}
 
-	const (
-		contractName = "OffRamp"
-		funcName     = "getExpectedNextSequenceNumbers"
-	)
+	res := make([]cciptypes.SeqNum, 0, len(chains))
+	for _, chain := range chains {
+		cfg, exists := cfgs[chain]
+		if !exists {
+			return nil, fmt.Errorf("source chain config not found for chain %d", chain)
+		}
+		if cfg.MinSeqNr == 0 {
```

Comment:
What if it's first message ever on a new chain?

Reply:
next expected is 1

Reply:
Can we already change names?! 😂 😮‍💨

Reply:
i'll fix in another pr